rfc/003-skip-values.md

```diff
@@ -0,0 +1,153 @@
+altered, with placeholder values removed before being persisted. This poses a problem if
+users later update their skip values, as any removed data would no longer be available in
+the MPI for future use.
+
+The second alternative is to apply specific rules during evaluation to determine whether a
+value should be used in a comparison or ignored as a placeholder. The main flaw with this
+approach is the lack of context. For example, if comparing the `FIRST_NAME` field and the
+incoming record has the value "John", this is typically a valid name for comparison.
+However, if the `LAST_NAME` is "Doe", it may be recognized as a placeholder. Performing the
+cleaning upfront allows for a comprehensive review of all data fields before evaluation,
+ensuring that additional context is considered.
+
+## Risks and Drawbacks
+
+The primary drawback of this approach is the added complexity, as it introduces an
+additional step to the process. Previously, the core steps for running linkage were:
+1. Data normalization
+2. Blocking
+3. Comparisons (also known as evaluation)
+4. Aggregation and prediction
+
+With the new approach, a "cleaning" step is added between steps 1 and 2. While the
+computational overhead of this additional step is minimal, the increased complexity is a
+concern. Each added step makes the system more challenging to evolve and harder for both
+users and developers to understand.
+
+## Implementation Plan
+
+For the purposes of this RFC, we will not be overly prescriptive about the implementation
+details. However, the work can be broadly divided into three tasks:
+1. A new `NAME` feature will be created, that will allow us to specify skip conditions
+    for the entirety of the name specified. (This likely won't be used for evaluation,
+    as its still preferable to compare the first and last names separately, but users
+    will have that option)
+2. Modify the existing Algorithm schema to include the new `skip_values`attribute,
+    along with parsing these values and storing the specified conditions.
+3. Implement a new cleaning step that takes the incoming data payload and a list of skip
+    conditions, then returns a copy of the data payload with placeholder values removed.
+    This cleaned copy will be used for blocking, evaluation, and aggregation, while the
+    original incoming payload will be retained for persistence.
+
+## Unresolved Questions
+
+A key question to consider is whether the concept of skip value conditions accounts for all
+possible scenarios. Do we need more complex rules that evaluate multiple attributes
+simultaneously? For example, should an `ADDRESS` of "123 Main St" be skipped only if the
+`ZIP` is also "99999"? While supporting rules of this complexity wouldn't require
+significant changes to the specification, it would increase the implementation complexity.
+We recommend pursuing this type of **and** logic only if there is a clear and specific need
+for it.
```

Comment:
Does NBS have skip values? I imagine theirs would give us insight into how complex the skip rules might be.

Reply:
Its a good thought, I brought it up with their team today.  I believe the answer is no, but they'll get back to us on this.